From f00bb0fef1b2644afaefbb828f42b7be812ac4a6 Mon Sep 17 00:00:00 2001
From: Omri Itach <omrii@marvell.com>
Date: Tue, 25 Jul 2017 15:45:33 +0300
Subject: [PATCH 1834/2241] dts: a8040: usb: enable USB Host0 on CP1 for OCP
 board

enable USB Host0 (USB2 with UTMI only), including VBUS support.

Change-Id: I62adcdec994a8c807d8e7bdba843169fd3fc9f22
Signed-off-by: Omri Itach <omrii@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/42122
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
---
 arch/arm64/boot/dts/marvell/armada-8040-ocp.dts | 25 +++++++++++++++++++++++++
 1 file changed, 25 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-8040-ocp.dts b/arch/arm64/boot/dts/marvell/armada-8040-ocp.dts
index 5bbc4af..97a0d76 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-ocp.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-ocp.dts
@@ -82,6 +82,12 @@
 
 	cpn-110-master {
 		config-space {
+			pinctrl@440000 {
+				xhci0_vbus_pins: xhci0-vbus-pins {
+					marvell,pins = "mpp47";
+					marvell,function = "gpio";
+				};
+			};
 			comphy0: comphy {
 				status = "okay";
 			};
@@ -111,6 +117,17 @@
 					status = "okay";
 				};
 			};
+			reg_usb3h0_vbus: usb3-vbus0 {
+				compatible = "regulator-fixed";
+				pinctrl-names = "default";
+				pinctrl-0 = <&xhci0_vbus_pins>;
+				regulator-name = "reg-usb3h0-vbus";
+				regulator-min-microvolt = <5000000>;
+				regulator-max-microvolt = <5000000>;
+				enable-active-high;
+				gpio = <&cpm_gpio1 15 GPIO_ACTIVE_HIGH>;
+				status = "okay";
+			};
 		};
 	};
 
@@ -119,6 +136,14 @@
 			comphy1: comphy {
 				status = "okay";
 			};
+			usb3h0_1: usb3@500000 {
+				usb-phy = <&usb3h0_phy>;
+				status = "okay";
+			};
+			usb3h0_phy: usb3_phy0 {
+				compatible = "usb-nop-xceiv";
+				vcc-supply = <&reg_usb3h0_vbus>;
+			};
 			gop {
 				emac0_1: mac0 {
 					phys = <&comphy0 4 COMPHY_SFI>;
-- 
2.7.4

